studio/components/interfaces/TableGridEditor/SidePanelEditor/ActionBar.tsx

```diff
@@ -44,6 +56,20 @@ const ActionBar: FC<ActionBarProps> = ({
           {applyButtonLabel}
         </Button>
       )}
+
+      {noApplyFunction && (
+        /**
+         * new solution to avoid using applyFunction
+         * this is necessary for handling loading by form itself
+         *
+         * this does not require he applyCallback() above
+         *
+         * author @mildtomato
+         */
+        <Button size="small" disabled={disableApply} loading={loading}>
+          {applyButtonLabel}
+        </Button>
+      )}
```

Comment:
could we do something like this? imo `noApplyFunction` is a little confusing, but we should still keep the comment about why we have 2 different button renders. This way we don't need the additional `noApplyFunction` prop too

```
{applyFunction ? (
  <Button onClick={onSelectApply}>{applyButtonLabel}</Button>
) : (
    <Button>{applyButtonLabel}</Button>
}
```

Reply:
This would mean the button would always render though, which I think you mentioned isn't useful for a couple of the SidePanels, as they don't need 'Submit' buttons